tkg/client/cluster_test.go

```diff
@@ -386,6 +387,43 @@ ova: []
 			})
 
 		})
+
+		When("ALLOW_LEGACY_CLUSTER is failed to set", func() {
+			BeforeEach(func() {
+				tkgConfigReaderWriter.GetCalls(func(key string) (string, error) {
+					configMap := populateConfigMap()
+					if val, ok := configMap[key]; ok {
+						return val, nil
+					}
+					if key == constants.ConfigVariableAllowLegacyCluster {
+						return "false", errors.Errorf("fake error")
+					}
+					return "192.168.2.1/16", nil
+				})
+
+				featureFlagClient.IsConfigFeatureActivatedStub = func(featureFlagName string) (bool, error) {
+					if featureFlagName == constants.FeatureFlagAllowLegacyCluster {
+						return false, nil
```

Comment:
**suggestion:** We should test for all combinations of:
- feature gate is true or false
- env var is absent, set to something we can't parse, set to true or false.

This is also why we should use a function to encapsulate our logic and return a clear value.

Reply:
@imikushin , about you mentioned tests, they have been done [here](https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/client/common_test.go#L97-L148).
So, I think we only need to test the situation where get ALLOW_LEGACY_CLUSTER env var failed.  Does this make sense?